xscen/aggregate.py

```diff
@@ -47,26 +47,26 @@ def climatological_mean(
     window: int = None,
     min_periods: int = None,
     interval: int = 1,
-    periods: list = None,
-    to_level: str = "climatology",
+    periods: Union[list[str], list[list[str]]] = None,
+    to_level: Optional[str] = "climatology",
```

Comment:
Having re-read the PR description, I still think that we don't lose much by keeping the `Optional`s around (these will be replaced with `xyz | None` in Python3.10 (or if we adopt `from __future__ import annotations`.

Reply:
I don't mind adding back those if you say that `Optional` should be everywhere. From my testing, PyCharm had no problem understanding that you can explicitly use `None`, when this was the default.

Reply:
PyCharm type checking isn't as exhaustive/extensive as `mypy`. Once we indicate to `pyupgrade` that we want to move to Python3.10 typing, this will look much nicer without any intervention needed on our end.

Reply:
Alright, in that case I'll go with `Optional` everywhere relevant.